app/models/course/assessment/submission.rb

```diff
@@ -10,7 +10,8 @@ class Course::Assessment::Submission < ActiveRecord::Base
 
   workflow do
     state :attempting do
-      event :finalise, transitions_to: :submitted
+      event :finalise, transitions_to: :graded, if: proc { |s| s.autograded? }
```

Comment:
Read the docs for workflow, what I can do is the following:
```ruby
workflow do
  state :attempting do
    # Transition to graded only if assessment is autograded. Otherwise it will seek the next finalise event.
    event :finalise, transitions_to: :graded, if: :autograded?
    event :finalise, transitions_to: :submitted
  end
  state :submitted do
...
```
Not sure if this is better? Because it does require the understanding of how the `workflow` gem works.